tracker.go

```diff
@@ -32,98 +34,160 @@ func (c *Config) SetDefault() {
 	c.thresholdStopRate = 4000
 	c.interval = 5 * time.Second
 	c.reportInterval = 15 * time.Second
-	c.captureMod = 1000
+	c.captureInterval = 200 * time.Millisecond
 	c.writer = os.Stdout
 }
 
 func (c *Config) NewTracker() *StackTracker {
-	return &StackTracker{
+	t := &StackTracker{
 		config: c,
 	}
+	//go t.intervalTester()
+	return t
 }
 
 type StackTracker struct {
-	config             *Config
-	intervalStartTime  time.Time
-	lastReport         time.Time
-	intervalStartCount int32
-	count              int32
-	traces             map[string]uint32
-}
+	config *Config
+	count  int32
 
-func (t *StackTracker) TrackCall(now time.Time) {
-	if t.intervalStartTime.IsZero() {
-		t.intervalStartTime = now
-	}
-	t.count += 1
+	tracesMutex  sync.RWMutex
+	traceChannel chan string
 
-	deltaT := now.Sub(t.intervalStartTime)
-	deltaC := t.count - t.intervalStartCount
+	lastCaptureMutex sync.Mutex
+	lastCapture      time.Time
+}
 
-	if deltaT > t.config.interval {
-		var rate float64 = float64(deltaC) * float64(time.Second) / float64(deltaT)
-		if t.isTracking() {
-			if rate < t.config.thresholdStopRate {
-				t.stopTracking(now)
-			}
-		} else {
-			if rate > t.config.thresholdStartRate {
-				t.startTracking(now)
-			}
+func (t *StackTracker) TrackCall(now time.Time) {
+	for {
+		count := atomic.LoadInt32(&t.count)
+		if atomic.AddInt32(&t.count, 1) == count+1 {
+			break
 		}
-
-		t.intervalStartTime = now
-		t.intervalStartCount = t.count
 	}
 
 	if t.isTracking() {
-		if deltaC%t.config.captureMod == 0 {
-			t.captureTrace()
+		t.lastCaptureMutex.Lock()
```

Comment:
In this specific case, even if thread 1 doesn't see the values written in thread 2, that just means that it will go ahead and enter the critical section, at which point, it *will* see the values written by thread 2.